presto-main/src/main/java/com/facebook/presto/metadata/FunctionMetadata.java

```diff
@@ -0,0 +1,61 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.metadata;
+
+import com.facebook.presto.spi.function.FunctionKind;
+import com.facebook.presto.spi.function.Signature;
+import com.facebook.presto.spi.type.TypeSignature;
+
+import java.util.List;
+
+import static com.google.common.base.Preconditions.checkArgument;
+import static java.util.Objects.requireNonNull;
+
+public class FunctionMetadata
+{
+    private final Signature signature;
```

Comment:
`variableArity` is never used in planning. `getFunctionKind` is added as a function. I thought about whether to store a `Signature` or store `name`, `functionKind`, `argumentTyes` and `returnType` separately and I don't have a strong opinion one way or the other right now. Using `signature` here purely because it's less words to type. We can change if there's evidence that not storing signature is the better option.